core/server/helm_release_test.go

```diff
@@ -82,6 +90,199 @@ func TestGetHelmRelease(t *testing.T) {
+						Name: "somesource",
+					},
+				},
+			},
+		},
+	}
+
+	opt := []client.PatchOption{
+		client.ForceOwnership,
+		client.FieldOwner("helmrelease-controller"),
+	}
+
+	g.Expect(k.Patch(ctx, &helmRelease, client.Apply, opt...)).To(Succeed())
+
+	helmRelease.Status.LastReleaseRevision = 1
+	helmRelease.ManagedFields = nil
+
+	g.Expect(k.Status().Patch(ctx, &helmRelease, client.Apply, opt...)).To(Succeed())
+
+	// Create helm storage.
+	storage := types.HelmReleaseStorage{
+		Name:     "",
+		Manifest: helmReleaseInventoryObjects(),
+	}
+
+	storageData, _ := json.Marshal(storage)
+
+	// Create secret.
+	storageSecret := v1.Secret{}
+	storageSecret.Namespace = ns1.Name
+	storageSecret.Name = fmt.Sprintf(
+		"sh.helm.release.v1.%s.v%v",
+		helmRelease.GetName(),
+		helmRelease.Status.LastReleaseRevision,
+	)
+	storageSecret.Data = map[string][]byte{
+		"release": []byte(base64.StdEncoding.EncodeToString(storageData)),
+	}
+	g.Expect(k.Create(ctx, &storageSecret)).To(Succeed())
+
+	response, err := c.GetHelmRelease(ctx, &pb.GetHelmReleaseRequest{
+		Name:      appName,
+		Namespace: ns1.Name,
+	})
+
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(response.HelmRelease.Name).To(Equal(appName))
+	g.Expect(response.HelmRelease.Namespace).To(Equal(ns1.Name))
+	g.Expect(response.HelmRelease.Inventory).To(HaveLen(2))
+}
+
+func TestGetHelmRelease_withInventoryCompressed(t *testing.T) {
+	g := NewGomegaWithT(t)
+	ctx := context.Background()
+
+	c, cleanup := makeGRPCServer(k8sEnv.Rest, t)
+	defer cleanup()
+
+	_, k, err := kube.NewKubeHTTPClientWithConfig(k8sEnv.Rest, "")
+	g.Expect(err).NotTo(HaveOccurred())
+
+	appName := "myapp" + rand.String(5)
+	ns1 := newNamespace(ctx, k, g)
+
+	// Create helm release.
+	helmRelease := helmv2.HelmRelease{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       helmv2.HelmReleaseKind,
+			APIVersion: helmv2.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: ns1.Name,
+		},
+		Status: helmv2.HelmReleaseStatus{
+			LastReleaseRevision: 2,
+		},
+		Spec: helmv2.HelmReleaseSpec{
+			Chart: helmv2.HelmChartTemplate{
+				Spec: helmv2.HelmChartTemplateSpec{
+					SourceRef: helmv2.CrossNamespaceObjectReference{
+						Kind: "GitRepository",
+						Name: "somesource",
+					},
+				},
+			},
+		},
+	}
+
+	opt := []client.PatchOption{
+		client.ForceOwnership,
+		client.FieldOwner("helmrelease-controller"),
+	}
+
+	g.Expect(k.Patch(ctx, &helmRelease, client.Apply, opt...)).To(Succeed())
+
+	helmRelease.Status.LastReleaseRevision = 1
+	helmRelease.ManagedFields = nil
+
+	g.Expect(k.Status().Patch(ctx, &helmRelease, client.Apply, opt...)).To(Succeed())
+
+	// Create helm storage.
+	storage := types.HelmReleaseStorage{
+		Name:     "",
+		Manifest: helmReleaseInventoryObjects(),
+	}
+
+	storageData, _ := json.Marshal(storage)
+
+	var compressed bytes.Buffer
+
+	compressor := gzip.NewWriter(&compressed)
+	_, _ = compressor.Write(storageData)
+	compressor.Close()
```

Comment:
any values we need to check here?

Reply:
Well, if we don't test compressed data, we can't be sure it's working.

Reply:
no i mean, does `Write` return anything we might care about? like an error

Reply:
it returns with the number of bytes written and an error if it failed, but it shouldn't fail, it's a write and we are not using filesystem io, buffer writing will not fail, if it does somehow, the test will fail anyway.

Reply:
The reason I put the `_, _ =` there because linter cried.